addons/base_address_extended/models/res_partner.py

```diff
@@ -17,123 +13,39 @@ class Partner(models.Model):
-                #maxsplit set to 1 to unpack only the first element and let the rest untouched
-                tmp = street_raw.split(separator, 1)
-                if previous_greedy in vals:
-                    # attach part before space to preceding greedy field
-                    append_previous, sep, tmp[0] = tmp[0].rpartition(' ')
-                    street_raw = separator.join(tmp)
-                    vals[previous_greedy] += sep + append_previous
-                if len(tmp) == 2:
-                    field_value, street_raw = tmp
-                    vals[field_name] = field_value
-            if field_value or not field_name:
-                previous_greedy = None
-                if field_name == 'street_name' and separator == ' ':
-                    previous_greedy = field_name
-                # select next field to find (first pass OR field found)
-                # [2:-2] is used to remove the extra chars '%(' and ')s'
-                field_name = re_match.group()[2:-2]
-            else:
-                # value not found: keep looking for the same field
-                pass
-            if field_name not in street_fields:
-                raise UserError(_("Unrecognized field %s in street format.", field_name))
-            previous_pos = re_match.end()
-
-        # last field value is what remains in street_raw minus trailing chars in street_format
-        trailing_chars = street_format[previous_pos:]
-        if trailing_chars and street_raw.endswith(trailing_chars):
-            vals[field_name] = street_raw[:-len(trailing_chars)]
-        else:
-            vals[field_name] = street_raw
-        return vals
-
-    def write(self, vals):
-        res = super(Partner, self).write(vals)
-        if 'country_id' in vals and 'street' not in vals:
-            self._inverse_street_data()
-        return res
-
-    def _formatting_address_fields(self):
-        """Returns the list of address fields usable to format addresses."""
-        return super(Partner, self)._formatting_address_fields() + self._get_street_fields()
-
-    def _get_street_fields(self):
-        """Returns the fields that can be used in a street format.
-        Overwrite this function if you want to add your own fields."""
-        return ['street_name', 'street_number', 'street_number2']
+            partner.update(tools.street_split(partner.street))
+
+    def _get_street_split(self):
+        self.ensure_one()
+        return {
+            'street_name': self.street_name,
+            'street_number': self.street_number,
+            'street_number2': self.street_number2
+        }
+
+    @api.onchange('city_id')
+    def _onchange_city_id(self):
+        if self.city_id:
+            self.city = self.city_id.name
+            self.zip = self.city_id.zipcode
+            self.state_id = self.city_id.state_id
+        elif self._origin:
+            self.city = False
+            self.zip = False
+            self.state_id = False
```

Comment:
I don't understand why `False` in case of `_origin`

Reply:
it is  from original base_address_city:
https://github.com/odoo/odoo/commit/d7af73dcb1b4ef3ada39bcafa7ad23972e487b45

Reply:
basically, what @poma-odoo said, and...
if the city_id is set, we copy the city_id field values to the partner fields,
but if you clear the city_id, only reset the partner fields if it's already in the db.